docs/design/dynamic-mig.md

```diff
@@ -0,0 +1,158 @@
+# NVIDIA GPU MPS and MIG dynamic slice plugin
+
+## Special Thanks
+
+This feature will not be implemented without the help of @sailorvii.
+
+## Introduction
+
+The NVIDIA GPU build-in sharing method includes: time-slice, MPS and MIG. The context switch for time slice sharing would waste some time, so we chose the MPS and MIG. The GPU MIG profile is variable, the user could acquire the MIG device in the profile definition, but current implementation only defines the dedicated profile before the user requirement. That limits the usage of MIG. We want to develop an automatic slice plugin and create the slice when the user require it.
+For the scheduling method, node-level binpack and spread will be supported. Referring to the binpack plugin, we consider the CPU, Mem, GPU memory and other user-defined resource.
+Volcano already have a [vgpu feature](https://github.com/Project-HAMi/volcano-vgpu-device-plugin) for NVIDIA devices after v1.9, it is done by using [hami-core](https://github.com/Project-HAMi/HAMi-core), which is a cuda-hacking library. But mig is also widely used across the world. A unified API for dynamic-mig and volcano-vgpu is needed.
+
+## Targets
+
+- CPU, Mem, and GPU combined schedule
+- GPU dynamic slice: Hami-core and MIG
+- Support node-level binpack and spread by GPU memory, CPU and Mem
+- A unified vGPU Pool different virtualization technics
+- Tasks can choose to use MIG, use HAMi-core, or use both.
+
+### Config maps
+- volcano-device-configMap
+This configmap defines the plugin configurations including resourceName, and MIG geometries, and node-level configurations.
+
+```yaml
+apiVersion: v1
+data:
+  volcano-device-share.conf: |
+    nvidia:
+      resourceCountName: volcano.sh/vgpu-number
+      resourceMemoryName: volcano.sh/vgpu-memory
+      resourceCoreName: volcano.sh/vgpu-cores
+      knownMigGeometries:
+      - models: [ "A30" ]
+        allowedGeometries:
+          - 
```

Comment:
Why is the yaml here so weird? There are two `-` in the list, should it be like this?
```yaml
knownMigGeometries:
- models: [ "A30" ]
   allowedGeometries:
     -  name: 1g.6gb
         memory: 6144
         count: 4
     -  name: 2g.12gb
         memory: 12288
         count: 2
     -  name: 4g.24gb
         memory: 24576
         count: 1
     - ....
 ```

Reply:
Can we describe what some of these fields mean and why they are configured the way they are?